### PR TITLE
fix: warn when environment variables referenced in config are not set

### LINKED ISSUE #1532: Environment variables inconsistency
### Describe the bug

I have the following configuration file snippet. I'm using docker compose and have tried both env_file and enviornment to set the variables.

The environment variables for security and alerting work as expected. However, the variables under the endpoint are not interpolated.

```
security:
  basic:
    username: ${AUTH_USER}
    password-bcrypt-base64: ${AUTH_BCRYPT}

alerting:
  slack:
    webhook-url: "https://hooks.slack.com/services/${SLACKHOOK}"

connectivity:
  checker:
    target: 1.1.1.1:53
    interval: 60s

default-endpoint: &defaults
  group: core
  interval: 120s
  client:
    timeout: 15s
  conditions:
    - "[STATUS] == 200"
    - "[CERTIFICATE_EXPIRATION] > 48h"
  alerts:
    - type: slack
      description: "healthcheck failed 2 times in a row"
      failure-threshold: 2
      send-on-resolved: true

endpoints:
  - name: API
    <<: *defaults
    group: infrastructure
    url: "https://api.site/staff/${KRECORD}/basic"
    headers:
      Authorization: "Bearer ${KTOKEN}"
```

### What do you see?

On the Gatus site I am prompted for user/pass. When a site has an issue an alert is sent to Slack. Indicating top three environment variables are working.

Watching the logs on api.site I see the request come through:

```
"GET /staff//basic HTTP/1.1" 401 179 "-" "Gatus/1.0"
```

Indicating KRECORD has not been expanded into the url of the request. The 401 indicating the authorization has also failed.

### What do you expect to see?

The following request on the api site and a nice green dashboard.

`
"GET /staff/12345678/basic HTTP/1.1" 200 179 "-" "Gatus/1.0"
`

### List the steps that must be taken to reproduce this issue

- add endpoints with environment variable to config.yaml
- set environment variables in .env file or under environment in compose file
- run gatus via docker compose

### Version

latest docker image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by yosofbadr**
## Summary

- Replace `os.ExpandEnv` with `os.Expand` + `os.LookupEnv` in `parseAndValidateConfigBytes` so that a warning is logged whenever an environment variable referenced in the configuration (including endpoint `url`, `headers`, and `body`) is not defined in the environment.
- Add tests verifying env var expansion works correctly in endpoint URL, headers, and body fields.
- Add test confirming undefined env vars produce an empty-string substitution (the existing behavior) while now also emitting a warning log.

This helps users diagnose issues like the one reported in #1532, where Docker Compose quoting (`- KTOKEN="value"`) caused the literal quotes to become part of the value, leading to YAML parse errors or unexpected substitution results. The warning makes it immediately clear which variable was not found.

Closes #1532

## Test plan

- [x] `TestParseAndValidateConfigBytesWithEnvVarInEndpointURLAndHeaders` -- verifies `${VAR}` in endpoint URL, Authorization header, and body are expanded
- [x] `TestParseAndValidateConfigBytesWithUndefinedEnvVarWarning` -- verifies undefined vars expand to empty string (and a warning is logged)
- [x] `TestParseAndValidateConfigBytesWithLiteralDollarSign` -- existing test still passes (literal `$$` handling unchanged)
- [x] Full `go test ./config/` suite passes

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -1846,6 +1846,62 @@ endpoints:
...
+    conditions:
+      - "[STATUS] == 200"
+`))
+	if err != nil {
+		t.Fatal("expected no error, got", err.Error())
+	}
+	if config == nil {
+		t.Fatal("config should not be nil")
+	}
+	// Undefined env vars should be replaced with empty string
```
Comment (TwiN) [inline config/config_test.go:1898]: Was this not already the case? Or was it just leaving the unset environment variable as a literal value?

**Comment by yosofbadr** [comment]
Good question @TwiN — yes, empty-string substitution was already the behavior of `os.ExpandEnv` (it wraps `os.Getenv` which returns `""` for unset variables). This test just locks in that behavior so the switch to `os.Expand` + `os.LookupEnv` doesn't accidentally regress it. The *new* behavior introduced by the PR is the `logr.Warnf` line — the substitution itself is unchanged.
